docs/release-notes/flink-1.9.md

```diff
@@ -0,0 +1,175 @@
+
+
+These release notes discuss important aspects, such as configuration, behavior,
+or dependencies, that changed between Flink 1.8 and Flink 1.9. It also provides an overview
+on known shortcoming or limitations with new experimental features introduced in 1.9.
+
+Please read these notes carefully if you are planning to upgrade your Flink version to 1.9.
+
+* This will be replaced by the TOC
+{:toc}
+
+## Known shortcomings or limitations for new features
+
+### New Table / SQL Blink runner
+
+TBD
+
+### Java 9 support
+
+Since Flink 1.9.0, Flink can now be compiled and run on Java 9. Note that certain components interacting
+with eternal systems (connectors, filesystems, metric reporters, etc.) may not work since the respective projects may
+have skipped Java 9 support.
+
+Related issues:
+- [FLINK-8033: JDK 9 support](https://issues.apache.org/jira/browse/FLINK-8033)
+
+## Deprecations and breaking changes
+
+### Failover strategies
+
+As a result of completion of fine-grained recovery ([FLIP-1](https://cwiki.apache.org/confluence/display/FLINK/FLIP-1+%3A+Fine+Grained+Recovery+from+Task+Failures)),
+Flink will now attempt to only restart tasks that are
+connected to failed tasks through a pipelined connection. By default, the `region` failover strategy is used.
+
+Users who were not using a restart strategy or have already configured a failover strategy should not be affected.
+Moreover, users who already enabled the `region` failover strategy, along with a restart strategy that enforces a
+certain number of restarts or introduces a restart delay, will see changes in behavior.
+The `region` failover strategy now correctly respects constraints that are defined by the restart strategy.
+
+Streaming users who were not using a failover strategy may be affected if their jobs are embarrassingly parallel or
+contain multiple independent jobs. In this case, only the failed parallel pipeline or affected jobs will be restarted. 
+
+Batch users may be affected if their job contains blocking exchanges (usually happens for shuffles) or the
+`ExecutionMode` was set to `BATCH` or `BATCH_FORCED` via the `ExecutionConfig`. 
+
+Overall, users should see an improvement in performance.
+
+Related issues:
+- [FLINK-13223: Set jobmanager.execution.failover-strategy to region in default flink-conf.yaml](https://issues.apache.org/jira/browse/FLINK-13223)
+- [FLINK-13060: FailoverStrategies should respect restart constraints](https://issues.apache.org/jira/browse/FLINK-13060)
+
+### Job termination via CLI
+
+With the support of graceful job termination with savepoints for semantic correctness
+([FLIP-34](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=103090212)), a few changes
+related to job termination has been made to the CLI.
+
+From now on, the `stop` command with no further arguments stops the job with a savepoint targeted at the
+default savepoint location (as configured via the `state.savepoints.dir` property in the job configuration),
```

Comment:
There's no default value for `state.savepoints.dir` according to our current code in `CheckpointingOptions`[L108](https://github.com/apache/flink/blob/428ce1b938813fba287a51bf86e6c52ef54453cb/flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java#L108), so I suggest to add an explicit instruction here to ask user to either add `-p` option when issue the command, or make sure to set the `state.savepoints.dir` property